src/SubscriberBase.h

```diff
@@ -83,6 +83,15 @@ class SubscriberBaseT : public Subscriber<T>,
       bool startExecutor = true)
       : ExecutorBase(executor, startExecutor), cancelled_(false) {}
 
+  virtual ~SubscriberBaseT() {
+    if (!cancelled_ && originalSubscription_) {
```

Comment:
Where is it guaranteed that the `SubscriberBaseT` will be strongly retained until it receives a terminating signal? `SubscriptionBase` doesn't guarantee this, for example.